### PR TITLE
[RFC] make rpcrequest on host start for UpdateRemotePlugins only continue of #5856

### PR DESCRIPTION
**Pull request by Shougo**
Based PR: https://github.com/neovim/neovim/pull/5856
It improves the asynchronous RPC call.
Note: It does not improve the synchronous RPC call like autocmd define in remote plugin.

For example, my tested deoplete initialization:

Before: 71ms(_deoplete()) + 21ms(Python check)
After: 2ms(_deoplete()) + 21ms(Python check)

Note:  You can skip the Python check by `g:python3_host_prog`.
So the loading time is almost 2ms!

https://github.com/neovim/neovim/pull/5856#issuecomment-310956062

It is the faster than Vim if_python initialization(almost 20ms)!

### CODE REVIEW

**Comment by Shougo** [comment]
More testers are needed for the branch.

**Comment by Shougo** [comment]
Why the CI is failed?

File: runtime/autoload/remote/define.vim [hunk 1]
```diff
@@ -169,14 +169,27 @@ function! remote#define#FunctionOnChannel(channel, method, sync, name, opts)
 endfunction
 
 
+function! remote#define#SafeRPCNofity(...) abort
```
Comment (justinmk) [inline runtime/autoload/remote/define.vim]: what's this? Should be a separate PR.

**Comment by justinmk** [comment]
The conflict wasn't resolved correctly. I am working on it now.

File: runtime/autoload/remote/define.vim (hunk 1 above)
Comment (justinmk) [inline runtime/autoload/remote/define.vim]: Even with this, the first call to `:UpdateRemotePlugins` fails.

File: runtime/autoload/remote/define.vim [hunk 2]
```diff
@@ -169,14 +169,27 @@ function! remote#define#FunctionOnChannel(channel, method, sync, name, opts)
...
+  if !exists('s:poll')
+    let s:poll = {}
+  endif
+  " check the process started correctly
+  if !get(s:poll, a:1, 0)
+    if rpcrequest(a:1, 'poll') ==# 'ok'
+      let s:poll[a:1] = 1
+    endif
+  endif
+  return call('rpcnotify', a:000)
```
Comment (justinmk) [inline runtime/autoload/remote/define.vim]: I don't understand the purpose of this. `s:poll` gets fill up but it only controls whether `rpcrequest(a:1, 'poll') ==# 'ok'` is checked. the result of that check is never used, only stored...

Comment (Shougo) [inline runtime/autoload/remote/define.vim]: The check is to wait the provider initialization.
If the check is not exists, the initial rpcnotify will be error.

File: runtime/autoload/remote/define.vim (hunk 1 above)
Comment (Shougo) [inline runtime/autoload/remote/define.vim]: I don't reproduce it...

**Comment by Shougo** [comment]
> The conflict wasn't resolved correctly. I am working on it now.

I don't see the PR.
It will be uploaded soon?

**Comment by justinmk** [comment]
I ran into a segfault which I need to fix before pushing this.

**Comment by Shougo** [comment]
OK.
